frontend_tests/node_tests/message_flags.js

```diff
@@ -148,6 +148,23 @@ run_test("read", (override) => {
         success: channel_post_opts.success,
     });
 
+    // Messages still not acked yet
+    const events = {};
+    const stub_delay = 100;
+    function set_timeout(f, delay) {
+        assert.equal(delay, stub_delay);
+        events.f = f;
+        events.timer_set = true;
+        return;
+    }
+    set_global("setTimeout", set_timeout);
```

Comment:
I think we had better ways to mock than `set_global`?

Reply:
We still want `set_global` for things like `setTimeout` that are clearly globals. This is the proper testing technique here.  Override `setTimeout` to make sure it's invoked correctly, and then capture the function for further testing.